lib/Lifters/FunctionLifter.cpp

```diff
@@ -1047,18 +1047,7 @@ void FunctionLifter::VisitInstruction(
     const remill::DecodingContext::ContextMap &mapper) {
   curr_inst = &inst;
 
-  // TODO(pag): Consider emitting calls to the `llvm.pcmarker` intrinsic. Figure
-  //            out if the `i32` parameter is different on 64-bit targets, or
-  //            if it's actually a metadata ID.
-
-  // Reserve space for an instrucion that will go into a delay slot, in case it
-  // is needed. This is an uncommon case, so avoid instantiating a new
-  // Instruction unless it is actually needed. The instruction instantition into
-  // this buffer happens via a placement new call later on.
-  std::aligned_storage<sizeof(remill::Instruction),
-                       alignof(remill::Instruction)>::type delayed_inst_storage;
-
-  remill::Instruction *delayed_inst = nullptr;
+  std::unique_ptr<remill::Instruction> delayed_inst = nullptr;
```

Comment:
Why is this an `unique_ptr`? Before it was local memory - `std::optional` therefore seems on the first glance as more natural replacement? `std::unique_ptr` is heap memory, so of course if we want the data to survive for longer this is the way (but it was not before?)

Reply:
yeah this probably isnt ideal... but the problem is all of the downstream methods take a pointer to an instruction then check if it's null, i suppose the correct thing here is to update every reference to the delayed instruction to be an optional